searchlib/src/vespa/searchlib/features/nowfeature.cpp

```diff
@@ -21,39 +19,37 @@ NowExecutor::execute(uint32_t) {
 }
 
 void
-NowBlueprint::visitDumpFeatures(const search::fef::IIndexEnvironment &,
-                                search::fef::IDumpFeatureVisitor &visitor) const
+NowBlueprint::visitDumpFeatures(const fef::IIndexEnvironment &, fef::IDumpFeatureVisitor &visitor) const
 {
     visitor.visitDumpFeature(getBaseName());
 }
 
 bool
-NowBlueprint::setup(const search::fef::IIndexEnvironment &,
-                    const search::fef::ParameterList &)
+NowBlueprint::setup(const fef::IIndexEnvironment &, const fef::ParameterList &)
 {
     describeOutput("out", "The timestamp (seconds since epoch) of query execution.");
     return true;
 }
 
-search::fef::Blueprint::UP
+fef::Blueprint::UP
 NowBlueprint::createInstance() const
 {
-    return search::fef::Blueprint::UP(new NowBlueprint());
+    return std::make_unique<NowBlueprint>();
 }
 
-search::fef::FeatureExecutor &
-NowBlueprint::createExecutor(const search::fef::IQueryEnvironment &env, vespalib::Stash &stash) const
+using namespace std::chrono;
+
+fef::FeatureExecutor &
+NowBlueprint::createExecutor(const fef::IQueryEnvironment &env, vespalib::Stash &stash) const
 {
     int64_t timestamp;
     const fef::Property &prop = env.getProperties().lookup(fef::queryproperties::now::SystemTime::NAME);
     if (prop.found()) {
         timestamp = atoll(prop.get().c_str());
     } else {
-        FastOS_Time now;
-        now.SetNow();
-        timestamp = (int64_t)now.Secs();
+        timestamp = duration_cast<seconds>(system_clock::now().time_since_epoch()).count();
```

Comment:
This is the wrong kind of seconds. We are losing precision by going from double to int. If we open the door by using 'using std::chrono; some places, we probably need a separate type for seconds as double...

Reply:
This one is actually correct. It expects seconds since 1970 as an integer.